queue_downloads.py

```diff
@@ -0,0 +1,301 @@
+        downloads (list[int]): List of download ids for each part.
+        monitor_sleep (int): Number of seconds to wait between each check.
+
+    Raises:
+        RuntimeError: If a status code other than 200 is returned.
+    """
+    url = base_url + "/topcat/user/downloads/status"
+    params = {"sessionId": session_id, "downloadIds": downloads}
+    response = requests.get(url=url, params=params)
+    if response.status_code != 200:
+        raise RuntimeError(response.text)
+    content = json.loads(response.content)
+    print(content)
+
+    while any([s in {"QUEUED", "PAUSED", "PREPARING", "RESTORING"} for s in content]):
+        headers = {"Authorization": f"Bearer {session_id}"}
+        requests.put(url=base_url + "/datagateway-api/sessions", headers=headers)
+        if response.status_code != 200:
+            raise RuntimeError(response.text)
+
+        sleep(monitor_sleep)
+        response = requests.get(url=url, params=params)
+        if response.status_code != 200:
+            raise RuntimeError(response.text)
+
+        content = json.loads(response.content)
+        print(content)
+    
+    print("All downloads complete")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        prog="queue_downloads",
+        description=(
+            "Submits DataGateway Download requests for a list of specific filepaths. "
+            "The list will be split into separate parts of up to 10,000 files for "
+            "performance reasons. Once submitted Downloads will be visible in the "
+            "DataGateway UI as usual."
+        ),
+    )
+    parser.add_argument(
+        "input_file",
+        type=str,
+        help=(
+            "File containing the full paths of all files to submit for download, "
+            "separated by newlines. The path should match the 'location' field "
+            "displayed in the DataGateway UI."
+        ),
+    )
+    parser.add_argument(
+        "--url",
+        type=str,
+        default="https://datagateway.diamond.ac.uk",
+        help="The url address of the DataGateway instance to submit requests to.",
+    )
+    parser.add_argument(
+        "-a",
+        "--authenticator",
+        type=str,
+        default="ldap",
+        help="The authentication mechanism to use for DataGateway login.",
+    )
+    parser.add_argument(
+        "-u",
+        "--username",
+        type=str,
+        required=True,
+        help="The username used for DataGateway login.",
+    )
+    parser.add_argument(
+        "-p",
+        "--password-file",
+        type=str,
+        help=(
+            "Location of file containing password for DataGateway login. If not "
+            "provided, the password will need to be provided by prompt."
+        ),
+    )
+    parser.add_argument(
+        "--download-name",
+        type=str,
+        help=(
+            "Custom file name for the download(s). If not set will default to the "
```

Comment:
Perhaps make it "file name/identifier" - it becomes the file name for a zip download but for recalls to filesystems it is just an identifier for the download.